guildwars2/database.py

```diff
@@ -326,13 +326,13 @@ def check(m):
         def consolidate_duplicates(items):
             unique_items = collections.OrderedDict()
             for item in items:
-                item_tuple = item["name"], item["rarity"]
+                item_tuple = item["name"], item["rarity"], item["type"]
```

Comment:
This will cause search not to consolidate items with same name and rarity but different type. This is unwanted as user can't see the item type. Could check if the itemgroup after user has selected one contains an item of type `UpgradeComponent`, and have it act as a flag whether to search upgrades or not

Reply:
pulled that out into custom field after user response